yt/yt/tests/integration/proxies/test_http_proxy.py

```diff
@@ -1154,3 +1177,52 @@ def test_ping_https_verify_false(self):
     def test_ping_verify_ca(self):
         rsp = requests.get(self._get_https_proxy_url() + "/ping", verify=self._get_ca_cert())
         rsp.raise_for_status()
+
+    @authors("khlebnikov")
+    def test_certificate_update(self):
+        proxy_cert, proxy_cert_key = self._get_proxy_cert_path()
+
+        old_fingerprint = get_certificate_fingerprint(proxy_cert)
+        print_debug("Old certificate fingerprint: {}", old_fingerprint)
+
+        def current_fingerprint():
+            return get_certificate_fingerprint(cert_content=get_server_certificate(self._get_https_proxy_address()))
+
+        assert current_fingerprint() == old_fingerprint
+
+        create_certificate(
+            ca_cert=self._get_ca_cert(),
+            ca_cert_key=self.Env.yt_config.ca_cert_key,
+            cert=proxy_cert,
+            cert_key=proxy_cert_key,
+            names=[self.Env.yt_config.fqdn, self.Env.yt_config.cluster_name],
+        )
+
+        new_fingerprint = get_certificate_fingerprint(proxy_cert)
+        print_debug("New certificate fingerprint: {}", new_fingerprint)
+
+        assert new_fingerprint != old_fingerprint
+
+        wait(lambda: current_fingerprint() == new_fingerprint)
+
+        rsp = requests.get(self._get_https_proxy_url() + "/ping", verify=self._get_ca_cert())
+        rsp.raise_for_status()
+
+    @authors("khlebnikov")
+    def test_certificate_broken_update(self):
+        proxy_cert, proxy_cert_key = self._get_proxy_cert_path()
+
+        with open(proxy_cert, 'rb') as f:
+            cert_content = f.read()
+
+        with open(proxy_cert, 'w') as f:
+            f.write('hello')
+
+        time.sleep(3)
```

Comment:
It is better to use `wait` function

Reply:
I dont'n see how to do that without adding structured logging events for that case. This 
seems too overengineed.

I've removed this test case. It's not so important anyway.